portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,299 @@
+ *    either a single or an array of contact data.
+ * 
+ * @returns {!Array} The array of "contact-entry" html nodes.
+ */
+function createContactEntries(contactData) {
+  const contactEntryTemplate = document.querySelector('#contact-entry');
+  const entries = Object.entries(contactData).map(
+    ([title, data]) => loadContactEntry(contactEntryTemplate.content.cloneNode(true), title, data)
+  );
+
+  return entries;
+}
+
+/**
+ * This function populates an empty "contact-entry" html node given the contact data. 
+ * 
+ * @param {!DocumentFragment} entry The empty contact-entry html node.
+ * @param {!string} title The title of the current contact-entry.
+ * @param {!(string | {text: string, link: string} | Array<string | {text: string, link: string}>)} data 
+ *    A single or an array of contact data, which could be a string or an object with text and a link.
+ * 
+ * @returns {!DocumentFragment} The populated entry.
+ */
+function loadContactEntry(entry, title, data) {
+  entry.querySelector('.title').append(title);
+
+  if (Array.isArray(data)) {
+    valueDivs = data.map(createContactEntryContent);
+    entry.querySelector('.content').append(...valueDivs);
+  } else {
+    const div = createContactEntryContent(data);
+    entry.querySelector('.content').append(div);
+  }
+
+  return entry;
+}
+
+/**
+ * This function creates either a div or an anchor element,
+ * based on the provided data.
+ * 
+ * @param {!(string | {text: string, link: string})} data 
+ *    Contact information to be used in the created element
+ * 
+ * @returns {!HTMLElement} The element containing contact information
+ */
+function createContactEntryContent(data) {
+  const node = document.createElement('div');;
+
+  if (typeof data === 'object') {
+    const { text, link } = data;
+
+    const a = document.createElement('a');
+    a.href = link;
+    a.append(text);
+
+    node.append(a);
+  } else {
+    node.append(data);
+  }
+
+  return node;
+}
+
+/**
+ * Processes and appends the given experience section information on the website body.
+ * 
+ * @param {!object} sectionData Dictionary containing the experience data to be shown. 
+ *    Its keys contain the name of each experience section, and its values are
+ *    an array of objects describing individual experiences.
+ */
+function loadSectionData(sectionData) {
+  const sectionContainer = document.querySelector('main');
+  const sections = createSections(sectionData);
+
+  sectionContainer.append(...sections);
+}
+
+/**
+ * This function converts each element in the given section data to 
+ * an html node based on the "section" template.
+ * 
+ * @param {!object} sectionData Dictionary containing the experience data to be shown. 
+ *    Its keys contain the name of each experience section, and its values are
+ *    an array of objects describing individual experiences.
+ * 
+ * @returns {!Array} The array of "section" html nodes.
+ */
+function createSections(sectionData) {
```

Comment:
This and the `createContactEntries` look a bit similar, could we try to merge them into a generic function?

Reply:
I will look into this. They definitely look extremely similar, but I can't think of a way to logically group that code together (what to name the function/variables, for example). It just so happens to be a very similar JSON structure.

Maybe it'll come to me with a bit more thinking 🤔

Reply:
With the slight restructuring, this repeated code is not there anymore. How does it look?